### PR TITLE
ccc: Add preliminary aarch64 and morello support

### PR DESCRIPTION
**Pull request by jrtc27**
Flags taken from cheribuild but otherwise untested.

Also fix the riscv64 description whilst here to match the riscv64 variants.

### CODE REVIEW

File: tools/ccc [hunk 1]
```diff
@@ -64,6 +67,9 @@ if [ $# -eq 0 ]; then
 	usage
 fi
 
+arch=$1
+shift
```
Comment (brooksdavis) [inline tools/ccc:71]: I think we need to move the setting of SYSROOT below here as it's currently:
```
SYSROOT=${SYSROOT:-${CHERIBUILD_SDK}/sysroot-riscv64-purecap}
```

Comment (jrtc27) [reply tools/ccc:71]: Oops, yes, I was in a FETT mindset where it's just called sysroot. I guess that should be `sysroot-${arch%%-*}-purecap` too?

Comment (brooksdavis) [reply tools/ccc:71]: I don't have a build around, but that's what I'd expect. There might also be some issues with finding the right clang if we end up having two llvm's around.

Comment (jrtc27) [reply tools/ccc:71]: Yeah... emphasis on "preliminary" :)

**Review by brooksdavis — APPROVED** [review]

File: tools/ccc (hunk 1 above)
Comment (brettferdosi) [reply tools/ccc:67]: The sysroots for Morello produced by cheribuild are `${CHERIBUILD_SDK}/sysroot-aarch64`, `${CHERIBUILD_SDK}/sysroot-morello-hybrid`, and `${CHERIBUILD_SDK}/sysroot-morello-purecap`.

Could the default value for `${SYSROOT}` be `${CHERIBUILD_SDK}/sysroot-${arch}` when `${arch}` is any of the riscv64 flavors or aarch64, and `${arch/aarch64/morello}` for morello hybrid and purecap?

Comment (jrtc27) [reply tools/ccc:67]: That's an annoying decision on cheribuild's part. For the sake of consistency I think they should be changed to all say aarch64.

**Review by brooksdavis — APPROVED** [review]

**Review by arichardson — APPROVED** [review]
LGTM, I should have a sysroot built soon so I can actually test it.

**Review by brettferdosi — APPROVED** [review]

**Pull request merged**